includes/controllers/class-admin.php

```diff
@@ -65,6 +73,34 @@ public function __construct( $args = [] ) {
 		parent::__construct( $args );
 	}
 
+	public function watermark_plugin( $request ) {
+		$license_key = 'test';
+		$file_path   = '/test/test.zip';
+
+		if ( file_exists( $file_path ) ) {
+
+			// Open the existing ZIP file
+			$zip = new \ZipArchive();
+
+			if ( $zip->open( $file_path ) === true ) {
+				$zip->setArchiveComment( $license_key );
```

Comment:
Is it easy to detect and remove for pirates, e.g. via the 7Zip or other software UI?

Reply:
The watermark should be as hidden as possible blended into the file or if not then embedded into the obfuscated code of one of the inner archive files.